lib/manageiq/api/common/middleware/web_server_metrics.rb

```diff
@@ -0,0 +1,53 @@
+module ManageIQ
+  module API
+    module Common
+      module Middleware
+        class WebServerMetrics
+          def initialize(app, options = {})
+            @app            = app
+            @metrics_prefix = options[:metrics_prefix] || "http_server"
+
+            require 'prometheus_exporter/client'
+            require 'prometheus_exporter/metric'
+
+            @counter   = PrometheusExporter::Client.default.register(:counter, "#{@metrics_prefix}_total", "The total number of HTTP requests handled by the Rack application.")
```

Comment:
I put all 3 of these metrics into a single Middleware because there is a bunch of setup in common.  Let me know if you think I should break it out.